fastestimator/network.py

```diff
@@ -656,37 +699,11 @@ def transform(self, data: Dict[str, Any], mode: str, epoch: int = 1) -> Dict[str
         Returns:
             (batch_data, prediction_data)
         """
-        self.load_epoch(mode, epoch, warmup=False)
-        data = to_tensor(data, target_type="tf")
-        data, prediction = self.run_step(data)
-        self.unload_epoch()
-        # handle tensorflow multi-gpu inferencing issue, it will replicate data on each device
-        if isinstance(tf.distribute.get_strategy(), tf.distribute.MirroredStrategy):
-            prediction = self._subsample_data(prediction, get_batch_size(data))
-        data.update(prediction)
-        return data
-
-    def _subsample_data(self, data: T, n: int) -> T:
-        """Subsample data by selecting the first n indices recursively.
-
-        Args:
-            data: The data to be subsampled.
-
-        Returns:
-            Subsampled data.
-        """
-        if isinstance(data, dict):
-            return {key: self._subsample_data(val, n) for (key, val) in data.items()}
-        elif isinstance(data, list):
-            return [self._subsample_data(val, n) for val in data]
-        elif isinstance(data, tuple):
-            return tuple([self._subsample_data(val, n) for val in data])
-        elif isinstance(data, set):
-            return set([self._subsample_data(val, n) for val in data])
-        elif hasattr(data, "shape") and list(data.shape) and data.shape[0] > n:
-            return data[0:n]
-        else:
-            return data
+        # Distribute multi-gpu data for processing
+        strategy = tf.distribute.get_strategy()
+        if isinstance(strategy, tf.distribute.MirroredStrategy):
+            data = next(iter(strategy.experimental_distribute_dataset(tf.data.Dataset.from_tensors(data))))
```

Comment:
this is creating a new instance of tf.Dataset every step.    it may impact our training speed.

Reply:
no, training uses run_step, not transform()